api/chain.yml

```diff
@@ -0,0 +1,307 @@
+      to: 
+        type: string
+      value:
+        type: number
+      gasPrice: 
+        type: number
+      gasLimit:
+        type: number
+      method: 
+        type: string
+      parameters: 
+        type: object
+      signature: 
+        type: string
+        readOnly: true
+  Tipset:
+    type: object
+    properties:
+      kind: 
+        type: string
+        enum: [tipset]
+        readOnly: true
+      id:
+        type: string
+        readOnly: true
+        example: 'zDPWYqFCyaQ9QJHr1qWZ19rFg3YRkcuXREkvpiHRJZhGN5T8SnyJ'
+      height:
+        type: number
+      blocks:
+        type: array
+        items:
+          $ref: '#/models/BlockHeader'  
+endpoints:
+  block_resource:
+    get:
+      operationId: getBlockById
+      summary: Find block by ID
+      description: Return information about the Block with the specified ID
+      tags:
+        - Blocks
+      parameters: 
+      - name: blockId
+        in: path
+        description: ID of the Block to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: '#/models/Block'
+        401:
+          $ref: 'api.yml#/components/responses/UnauthorizedError'
+  executed_message_resource:
+    get:
+      operationId: getExecutedMessageById
+      summary: Find ExecutedMessage by ID
+      description: Return information about the ExecutedMessage with the specified ID. An executed Message is one that has been included in a valid block in the chain. It includes return value and status information.
+      tags:
+        - Messages
+      parameters: 
+      - name: executedMessageId
+        in: path
+        description: ID of the executed message to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: '#/models/ExecutedMessage'
+        401:
+          $ref: 'api.yml#/components/responses/UnauthorizedError'
+  message_resource:
+    get:
+      operationId: getMessageById
+      summary: Find Message by ID
+      description: Return information about the Message with the specified ID
+      tags: 
+        - nessages
+      parameters: 
+      - name: messageId
+        in: path
+        description: ID of the message to fetch
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Success
+          content:
+            application/json:
+              schema:
+                $ref: '#/models/Message'
+        401:
+          $ref: 'api.yml#/components/responses/UnauthorizedError'
+  messages_collection:
+    post:
+      operationId: createMessage
+      summary: Send a Message
+      description: This method sends a new Message to an Actor recipient.
```

Comment:
We'll have to be a bit more explicit here about whether the message is already signed, or is to be signed by the node (and if so, with what key), etc, etc. Ok to pick just one answer for now, but it should be clear.

Reply:
It's obviously not impossible for the calling code to have already signed the message, but it's not exactly simple. Maybe it would be appropriate to specify a different content type that takes a base64 encoded binary representation of a message for signed messages and reserve the JSON content type for unsigned messages?